searx/engines/qwant.py

```diff
@@ -118,15 +118,10 @@ def response(resp):
 
     # check for an API error
     if search_results.get('status') != 'success':
-        msg = ",".join(
-            data.get(
-                'message',
-                [
-                    'unknown',
-                ],
-            )
-        )
-        raise SearxEngineAPIException('API error::' + msg)
+        error_code = data.get('error_code')
+        if error_code == 24:
+            raise SearxEngineTooManyRequestsException
+        raise SearxEngineAPIException(error_code)
```

Comment:
If error_code != 24 .. isn't it better we return the (old) `msg` string (`SearxEngineAPIException('API error::' + msg)`?

Reply:
I assumed the `msg` was outdated, I never got a `msg` when testing :unamused:

Reply:
@jazzzooo I tested with: https://api.qwant.com/v3/search/web?q=foo&locale=de_DE&safesearch=0&count=10&offset=39

```
{'message': ['offset must be a factor of 10'], 'error_code': 1}
```

While I was testing the above I see we have also other issues with this engine (in the request function) .. if it is OK for you I will squash my fixes in your commit.

Reply:
@return42 yeah no problem :+1: